debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java

```diff
@@ -275,10 +280,25 @@ public String getValue() {
         }
 
         public SchemaNameAdjuster createAdjuster() {
-            if (this == SchemaNameAdjustmentMode.AVRO) {
-                return SchemaNameAdjuster.create();
+            switch (this) {
+                case AVRO:
+                    return SchemaNameAdjuster.AVRO;
+                case AVRO_UNICODE:
+                    return SchemaNameAdjuster.AVRO_UNICODE;
+                default:
+                    return SchemaNameAdjuster.NO_OP;
+            }
+        }
+
+        public SchemaNameAdjuster createFieldNameAdjuster() {
```

Comment:
@harveyyue 
Should not this return just `FieldNamer`? It seems to me now that we are mixing two things together in the `SchemaNameAdjuster`. This is not wrond as such but the name and JavaDoc of the class implies soemthing different and in fact only half is always used.

What is IMHO correct is to have a `SchemaAdjuster`. The instance of it will be created in `CommonConnectorConfig` constructor form both options schema adjustment mode and fild name adjustment mode. And then there will be a single method `nameAdjuster()` that will return this instance and be used both for schema and field name mapping? WDYT?